src/packcc.c

```diff
@@ -934,10 +940,38 @@ static void char_array__add(char_array_t *array, char ch) {
     array->buf[array->len++] = ch;
 }
 
+static void char_array__append(char_array_t *array, const char *s, size_t len) {
+    if (array->max <= array->len + len) {
```

Comment:
`array->max < array->len + len` is the right inequality (not `<=`).

P.S. As for `char_array__add()`, `array->max <= array->len` is right because it is equivalent to `array->max < array->len + 1`.

Reply:
You're right, I forgot to change it. Fixed now (even though this function might not be needed anyway).